## МЕТОДЫ АНАЛИЗА ДАННЫХ И МАШИННОГО ОБУЧЕНИЯ

### ДОМАШНЕЕ ЗАДАНИЕ №1 : "ПЕРЕБОР ПАРАМЕТРОВ АЛГОРИТМОВ"

Кожевников Д.А. 



ИМПОРТ БИБЛИОТЕК

In [4]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as ex
import warnings

warnings.filterwarnings('ignore')

ЗАГРУЗКА И ОПИСАНИЕ ДАННЫХ

In [9]:
df = pd.read_csv("/home/dmitry/fpmi_data_science/Homework_1/data/row/heart.csv")
df.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

ПЕРЕМЕННЫЕ: <br>

`age` - Age of the patient (возраст пациента, лет) <br>
`sex` - Sex of the patient (пол пациента) <br>
`cp` - Chest pain type (тип боли в груди):
- 0 = Typical Angina (типичная стенокардия),
- 1 = Atypical Angina (атипичная стенокардия),
- 2 = Non-anginal Pain (боль несвязанная со стенокардией),
- 3 = Asymptomatic (бессимптомная) <br>

`trtbps` - Resting blood pressure (in mm Hg) (кровяное давление в состоянии покоя в мм. рт. ст.)<br>
`chol` - Cholestoral in mg/dl fetched via BMI sensor (уровень холестирина в мг/децилитр, определенный с помощью датчика BMI)<br>
`fbs` - Fasting blood sugar > 120 mg/dl (уровень сахара в крови натощак):
- 1 = True, 
- 0 = False <br>

`restecg` - Resting electrocardiographic results (результаты электрокардиографии в состоянии покоя):
- 0 = Normal (обычная),
- 1 = ST-T wave normality (нормальность волны ST-T),
- 2 = Left ventricular hypertrophy (гипертрофия левого желудочка)<br>

`thalachh` - Maximum heart rate achieved (Максимальная достигнутая частота сердечных сокращений)<br>
`oldpeak` - Previous peak (предыдущий пик)<br>
`slp` - Slope (наклон)<br>
`caa` - Number of major vessels (количество крупных сосудов)<br>
`thall` - Thalium Stress Test result (результаты стресс-теста на таллий) ~ (0,3)<br>
`exng` - Exercise induced angina:
- 1 = Yes, 
- 0 = No <br>

`output` - Target variable (целевая переменная):<br>
- 0 = less chance of heart attack,
- 1 = more chance of heart attack.

РАЗВЕДОЧНЫЙ АНАЛИЗ ДАННЫХ